lib/Plugin/Abstract.php

```diff
@@ -442,7 +442,11 @@ final public function resolveError($rawData)
      */
     final public function addEvent($method, array $data, $delayTime = NULL)
     {
-        // TODO
+        if (is_int($delayTime)) {
+            $this->middleware->addEventQueue([$method, new Varien_Object($data), $delayTime]);
+        } else {
+           $this->$method(new Varien_Object($data));
```

Comment:
Very good! But, to most accurately represent what happens on production the event should be added to the queue even when there is no `$delayTime`.

Reply:
Makes sense now.